### PR TITLE
Add XOR-gate tree to treegates

### PR DESCRIPTION
**Pull request by karthisrinivasan**

### CODE REVIEW

File: std/gates/treegates.act [hunk 1]
```diff
@@ -330,6 +330,41 @@ defproc sigbuf_gen (bool in; bool out[N])
...
+    { N > 0 : "Need positive no. of inputs to gate" };
+
+    [ N=1 -> in[0] = out; 
+    []N=2 -> xor2 x(in,out);
+    []N>2 -> bool outx[2]; 
+            [ N%2=0 ->  xortree <N/2>       x0(in[0..(N/2)-1],outx[0]);
+                        xortree <N/2>       x1(in[(N/2)..N-1],outx[1]);
+                        xor2 x(outx, out);
+            []N%2=1 ->  xortree <N/2>       x0(in[0..(N/2)-1],outx[0]);
+                        xortree <(N/2)+1>   x1(in[(N/2)..N-1],outx[1]);
```
Comment (rmanohar) [inline std/gates/treegates.act]: The selection can be eliminated by changing the second xortree to xortree<N-(N/2)>

Comment (karthisrinivasan) [reply std/gates/treegates.act]: Fixed, thanks! :)

File: std/gates/treegates.act [hunk 2]
```diff
@@ -330,6 +330,37 @@ defproc sigbuf_gen (bool in; bool out[N])
...
+    [keeper=2] ~in[1] & ~_in0 | ~_in1 & ~in[0] -> out+
+     _in1 & _in0 | in[1] & in[0] -> out-
+   }
+   sizing { _in0{-1}; _in1{-1}; out{-1} }
+}
+
+/*
+ * Build an XOR-gate tree with optional inversion of final output
+ */
+template <pint N; pbool invert>
```
Comment (rmanohar) [inline std/gates/treegates.act]: It looks like `invert` is an unused parameter?

Comment (karthisrinivasan) [reply std/gates/treegates.act]: Oops..
Fixed it

**Pull request merged**